src/build_vehicle_gui.cpp

```diff
@@ -1722,7 +1728,14 @@ struct BuildVehicleWindow : Window {
 
 	void OnPaint() override
 	{
-		this->GenerateBuildList();
+		auto old_num_hidden_engines = this->num_hidden_engines;
+		auto sel_id = this->GenerateBuildList();
```

Comment:
Why not `uint` and `EngineID` to match? I know it's a matter of personal opinion whether auto (and var in C#) is more or less readable than explicit types, but in code reviews without Intellisense it's definitely less readable. 🙂 Same goes for the counterpart in the autoreplace GUI.